Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift

```diff
@@ -39,16 +39,31 @@ extension SyntaxStringInterpolation {
 
 public protocol HasTrailingCodeBlock {
   var body: CodeBlockSyntax { get set }
+
+  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) throws
 }
 
-public extension HasTrailingCodeBlock where Self: SyntaxExpressibleByStringInterpolation {
-  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) {
-    self = "\(signature) {}"
+public extension HasTrailingCodeBlock where Self: StmtSyntaxProtocol {
+  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) throws {
+    let stmt = StmtSyntax("\(signature) {}")
+    guard let castedStmt = stmt.as(Self.self) else {
+      throw SyntaxStringInterpolationError.producedInvalidNodeType(expectedType: IfStmtSyntax.self, actualNode: stmt)
+    }
+    self = castedStmt
     self.body = CodeBlockSyntax(statements: bodyBuilder())
   }
 }
 
-extension CatchClauseSyntax: HasTrailingCodeBlock {}
+extension CatchClauseSyntax: HasTrailingCodeBlock {
+  public init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) throws {
```

Comment:
`signature` seems weird to use here (and for the others).

Reply:
What do you think about `header`? Since the rest is the body and HTML has header and body? Or do you have better ideas?

Reply:
I like that better than `signature`, yeah.